docs/topics/http/urls.txt

```diff
@@ -231,8 +231,11 @@ default argument for ``num``, ``"1"``. If the second pattern matches,
 Performance
 ===========
 
-Each regular expression in a ``urlpatterns`` is compiled the first time it's
-accessed. This makes the system blazingly fast.
+The URL function processes regular expressions in the ``urlpatterns`` list
+which is compiled upon the first http request. Subsequent requests use the
+cached configuration via the url resolver. Reversing URLs tends to be
+slower compared to routing depending on the complexity of the regular
```

Comment:
I think this comparison between `resolve()` and `reverse()` is a bit of a moot point, since resolving usually happens once per request, and reversing may happen any number of times in a request, but it is also not true in general. The linear time complexity of checking each pattern sequentially has quite a big impact on the performance of `resolve()`. Reversing on the other hand can look up the pattern in constant time, and the complexity of the regex doesn't seem to have much of an impact. Includes are flattened and regexes concatenated for the purposes of reverse, so that doesn't have much of an impact either. My tests suggest that if you have to check around 10 regexes or more, resolving is actually slower than reversing. 

Some points that might be interesting to discuss:
* Using multiple patterns with the same view name means that reverse will have to check each pattern, with worst-case time complexity scaling linearly with the number of patterns. For performance, it is better to use unique names for each pattern.
* The same holds true when passing the actual view function. If a view function has multiple related patterns, using the view name will yield better performance. 
* Resolving an url that is earlier in the tree of url patterns will be faster. The time scales more or less linear with the number of patterns that has to be checked. This is depth-first (included patterns are checked before the next pattern in the list), but;
* Any include pattern that does not match the url will skip the entire subtree of patterns. 
* For optimal performance, it is best to give each include a unique prefix, and to nest all patterns with a common prefix in an include.

I wouldn't go into too much detail in the docs. For most projects these will be micro-optimizations, if the difference is noticeable at all. I think for both resolving and reversing, the number of checked patterns (for resolving) or reversed urls (for reversing) has to be in the 1,000-10,000 range before the effect on an average request-response cycle becomes somewhat measurable.